spec/types/bmp_string_spec.rb

```diff
@@ -0,0 +1,68 @@
+# coding: utf-8
+# frozen_string_literal: true
+
+require_relative '../spec_helper'
+
+describe RASN1::Types::BmpString do
+  describe '.type' do
+    it 'gets ASN.1 type' do
+      expect(described_class.type).to eq('BmpString')
+    end
+  end
+
+  describe '#initialize' do
+    it 'creates a subjectString with default values' do
+      subject = described_class.new
+      expect(subject).to be_primitive
+      expect(subject).to_not be_optional
+      expect(subject.asn1_class).to eq(:universal)
+      expect(subject.default).to eq(nil)
+    end
+  end
+
+  describe '#to_der' do
+    it 'generates a DER string from UTF-8 input' do
+      subject = described_class.new
+      subject.value = (+'Α').force_encoding('UTF-8')
+      expect(subject.to_der).to eq("\x1E\x02\x03\x91".b)
+    end
+
+    it 'generates a DER string from UTF-16BE input' do
+      subject = described_class.new
+      subject.value = (+"\x03\x91").force_encoding('UTF-16BE')
+      expect(subject.to_der).to eq("\x1E\x02\x03\x91".b)
+    end
+
+    it 'generates a DER string according to ASN.1 class' do
+      subject = described_class.new(class: :context)
+      subject.value = 'abc'
+      expect(subject.to_der).to eq("\x9E\x06\x00a\x00b\x00c".b)
+    end
+
+    it 'generates a DER string according to default' do
+      subject = described_class.new(default: 'NOP', octet_length: 22)
```

Comment:
I grabbed `octet_length` from the IA5String test - but I can't find any references to it elsewhere 👀

Reply:
this is an unknown option. I do not remember why i did that...

Or maybe a copy-paste from bitstring with bit replaced by octet...

Reply:
Please remove it.

Reply:
I've updated and removed this now :+1: